tf_explain/core/vanilla_gradients.py

```diff
@@ -1,13 +1,29 @@
 """
 Core Module for Vanilla Gradients
 """
+from warnings import warn
 import tensorflow as tf
 
 from tf_explain.utils.display import grid_display
 from tf_explain.utils.image import transform_to_normalized_grayscale
 from tf_explain.utils.saver import save_grayscale
 
 
+UNSUPPORTED_ARCHITECTURE_WARNING = "\
+Unsupported model architecture for VanillaGradients. The last two layers of \
+the model should be: a layer which computes class scores with no activation, \
+followed by an activation layer."
+
+_activation_layer_classes = (
+    tf.keras.layers.Activation,
+    tf.keras.layers.LeakyReLU,
+    tf.keras.layers.PReLU,
+    tf.keras.layers.ReLU,
+    tf.keras.layers.Softmax,
+    tf.keras.layers.ThresholdedReLU,
+)
```

Comment:
Maybe you can check dynamically that the layer is in `tf.keras.activations`. So we don't have to store all the activation types on our side

Reply:
I don't think that would work because the things in `tf.keras.activations` are activation functions rather than layers. Unfortunately there isn't really anything to distinguish an activation layer in tf.keras from any other type of layer...

But I assume the vast majority of models people will be applying Vanilla Gradients to will have softmax as the last activation function so as long as we have `Activation` and `Softmax` on this list, I think we will catch most cases.